components/script_injector/renderer/script_injector_render_frame_observer.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/script_injector/renderer/script_injector_render_frame_observer.h"
+
+#include <string>
+#include <utility>
+
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/common/associated_interfaces/associated_interface_registry.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_script_source.h"
+
+namespace script_injector {
+
+ScriptInjectorRenderFrameObserver::ScriptInjectorRenderFrameObserver(
+    content::RenderFrame* render_frame)
+    : content::RenderFrameObserver(render_frame) {
+  render_frame->GetAssociatedInterfaceRegistry()
+      ->AddInterface<script_injector::mojom::ScriptInjector>(
+          base::BindRepeating(&ScriptInjectorRenderFrameObserver::BindReceiver,
+                              base::Unretained(this)));
+}
+
+void ScriptInjectorRenderFrameObserver::OnDestruct() {
+  delete this;
+}
+
+void ScriptInjectorRenderFrameObserver::BindReceiver(
+    mojo::PendingAssociatedReceiver<script_injector::mojom::ScriptInjector>
+        pending_receiver) {
+  receiver_.reset();
+  receiver_.Bind(std::move(pending_receiver));
+}
+
+ScriptInjectorRenderFrameObserver::~ScriptInjectorRenderFrameObserver() =
+    default;
+
+void ScriptInjectorRenderFrameObserver::RequestAsyncExecuteScript(
+    int32_t world_id,
+    const std::u16string& script,
+    bool user_activation,
+    bool want_result,
+    bool await_promise,
+    RequestAsyncExecuteScriptCallback callback) {
+  blink::WebScriptSource web_script_source =
+      blink::WebScriptSource(blink::WebString::FromUTF16(script));
+
+  render_frame()->GetWebFrame()->RequestExecuteScript(
+      world_id, {&web_script_source, 1u},
```

Comment:
`{}` => `base::make_span`

Reply:
I've seen both in Chromium code base, any particular reason to use one over the other?

Reply:
I only see one usage of `{}` in older code. This does an implicit conversion from a C array to a span so `make_span` is preferred